src/openrct2/drawing/Drawing.Sprite.cpp

```diff
@@ -442,8 +448,10 @@ void FASTCALL gfx_draw_sprite_palette_set_software(
         zoomed_dpi.width = dpi->width >> 1;
         zoomed_dpi.pitch = dpi->pitch;
         zoomed_dpi.zoom_level = dpi->zoom_level - 1;
+
+        const auto& spriteCoords = ScreenCoordsXY({ x >> 1, y >> 1 });
         gfx_draw_sprite_palette_set_software(
-            &zoomed_dpi, imageId.WithIndex(imageId.GetIndex() - g1->zoomed_offset), x >> 1, y >> 1, paletteMap);
+            &zoomed_dpi, imageId.WithIndex(imageId.GetIndex() - g1->zoomed_offset), spriteCoords, paletteMap);
```

Comment:
```suggestion
            &zoomed_dpi, imageId.WithIndex(imageId.GetIndex() - g1->zoomed_offset), {spriteCoords.x >> 1, spriteCoords.y >> 1 }, paletteMap);
```

(This will probably break formatting)